client/src/main/java/org/apache/rocketmq/client/impl/producer/DefaultMQProducerImpl.java

```diff
@@ -264,12 +280,20 @@ public void checkTransactionState(final String addr, final MessageExt msg,
 
             @Override
             public void run() {
-                TransactionListener transactionCheckListener = DefaultMQProducerImpl.this.checkListener();
-                if (transactionCheckListener != null) {
+                TransactionCheckListener transactionCheckListener = DefaultMQProducerImpl.this.checkListener();
+                TransactionListener transactionListener = getCheckListener();
+                if (transactionCheckListener != null || transactionListener != null) {
                     LocalTransactionState localTransactionState = LocalTransactionState.UNKNOW;
                     Throwable exception = null;
                     try {
-                        localTransactionState = transactionCheckListener.checkLocalTransaction(message);
+                        if (transactionCheckListener != null) {
+                            localTransactionState = transactionCheckListener.checkLocalTransactionState(message);
+                        } else if (transactionListener != null) {
+                            log.info("Used new check API in transaction message");
```

Comment:
Better to use log.debug，otherwise it may cause too much log info

Reply:
got it.